test/Atmos/EDMF/edmf_kernels.jl

```diff
@@ -501,6 +560,86 @@ function atmos_source!(
     # covariance microphysics sources should be applied here
 end;
 
+function compute_ρa_up(m, state, aux)
+    # Aliases:
+    turbconv = m.turbconv
+    gm = state
+    up = state.turbconv.updraft
+    N_up = n_updrafts(turbconv)
+    a_min = turbconv.subdomains.a_min
+    a_max = turbconv.subdomains.a_max
+    # in future GCM implementations we need to think about grid mean advection
+    ρa_up = vuntuple(N_up) do i
+        gm.ρ * enforce_unit_bounds(up[i].ρa / gm.ρ, a_min, a_max)
+    end
+    return ρa_up
+end
+
+function flux(::Advect{up_ρa{i}}, m, state, aux, t, ts, direction) where {i}
+    up = state.turbconv.updraft
+    ẑ = vertical_unit_vector(m, aux)
+    return up[i].ρaw * ẑ
+end
+function flux(::Advect{up_ρaw{i}}, m, state, aux, t, ts, direction) where {i}
+    up = state.turbconv.updraft
+    ẑ = vertical_unit_vector(m, aux)
+    ρa_up = compute_ρa_up(m, state, aux)
+    return up[i].ρaw * up[i].ρaw / ρa_up[i] * ẑ
+end
+function flux(
+    ::Advect{up_ρaθ_liq{i}},
+    m,
+    state,
+    aux,
+    t,
+    ts,
+    direction,
+) where {i}
+    up = state.turbconv.updraft
+    ẑ = vertical_unit_vector(m, aux)
+    ρa_up = compute_ρa_up(m, state, aux)
+    return up[i].ρaw / ρa_up[i] * up[i].ρaθ_liq * ẑ
+end
+function flux(
+    ::Advect{up_ρaq_tot{i}},
+    m,
+    state,
+    aux,
+    t,
+    ts,
+    direction,
+) where {i}
+    up = state.turbconv.updraft
+    ẑ = vertical_unit_vector(m, aux)
+    ρa_up = compute_ρa_up(m, state, aux)
+    return up[i].ρaw / ρa_up[i] * up[i].ρaq_tot * ẑ
+end
+
+function flux(::Advect{en_ρatke}, m, state, aux, t, ts, direction)
+    en = state.turbconv.environment
+    env = environment_vars(state, aux, n_updrafts(m.turbconv))
+    ẑ = vertical_unit_vector(m, aux)
+    return en.ρatke * env.w * ẑ
+end
+function flux(::Advect{en_ρaθ_liq_cv}, m, state, aux, t, ts, direction)
+    en = state.turbconv.environment
+    env = environment_vars(state, aux, n_updrafts(m.turbconv))
+    ẑ = vertical_unit_vector(m, aux)
+    return en.ρaθ_liq_cv * env.w * ẑ
+end
+function flux(::Advect{en_ρaq_tot_cv}, m, state, aux, t, ts, direction)
+    en = state.turbconv.environment
+    env = environment_vars(state, aux, n_updrafts(m.turbconv))
+    ẑ = vertical_unit_vector(m, aux)
+    return en.ρaq_tot_cv * env.w * ẑ
+end
+function flux(::Advect{en_ρaθ_liq_q_tot_cv}, m, state, aux, t, ts, direction)
+    en = state.turbconv.environment
+    env = environment_vars(state, aux, n_updrafts(m.turbconv))
+    ẑ = vertical_unit_vector(m, aux)
+    return en.ρaθ_liq_q_tot_cv * env.w * ẑ
+end
+
 # # in the EDMF first order (advective) fluxes exist only in the grid mean (if <w> is nonzero) and the uprdafts
 function flux_first_order!(
```

Comment:
what does this mean? We have advective fluxes for all prognostic variables in the environment as well.

Reply:
What does what mean? The comment `in the EDMF first order ...`? I'm not sure exactly, @yairchn?

Reply:
I think this (en.ρaθ_liq_q_tot_cv * env.w * ẑ) is the first order flux of covariances in the environment  - the third term on the RHS of this equation in Cohen et al 2020:

<img width="407" alt="Screen Shot 2020-12-05 at 6 58 01 PM" src="https://user-images.githubusercontent.com/24443883/101270214-d618c100-372b-11eb-90e0-1a5965a58479.png">

Reply:
I think this (en.ρaθ_liq_q_tot_cv * env.w * ẑ) is the first order flux of covariances in the environment - the third term on the RHS of this equation in Cohen et al 2020:

<img width="407" alt="Screen Shot 2020-12-05 at 6 58 01 PM" src="https://user-images.githubusercontent.com/24443883/101270255-2e4fc300-372c-11eb-8775-17c7c073fd66.png">

Reply:
I think we can keep behavior-changing adjustments to a separate PR. For now, this is just changing the specification.